File: Java/MergeSort.java
package Java;

public class MergeSort {

    public static void main(String[] args) {
        int []nums = {5,2,6,1,4,-1,2,22,12,1,-10,-13,-32};
        mergeSort(nums,0,nums.length - 1);
        for(int i : nums){
            System.out.print(i + " ");
        }
    }

    public static void mergeSort(int []nums, int left, int right){
        if(left < right){
            int mid = (right + left) / 2;
            mergeSort(nums,left,mid);
            mergeSort(nums,mid + 1,right);
            merge(nums,left,mid,right);
        }
    }

    public static void merge(int []nums,int start,int mid,int end){
        int n1 = mid - start + 1, n2 = end - mid;
        int []L = new int[n1];
        int []R = new int[n2];
        // dividing array into left and right
        for(int i = 0;i < n1;i++)
            L[i] = nums[start + i];

        for(int j = 0;j < n2;j++)
            R[j] = nums[mid + j + 1];

        // merging the two left and right sorted arrays
        int i = 0,j = 0;
        int k = start;
        while(i < n1 && j < n2){
            if(L[i] <= R[j])
                nums[k++] = L[i++];
            else{
                nums[k++] = R[j++];
            }

        }
        while(i < n1){
            nums[k++] = L[i++];
        }

        while(j < n2){
            nums[k++] = R[j++];
        }

    }


}

File: PHP/merge-sort.php
<?php
    // --------------- Merge Sort Algo ----------------------
    function mergesort(&$arr, $left, $right) {
        if ($left < $right) { 
            $mid = $left + (int)(($right - $left)/2);
            mergesort($arr, $left, $mid);
            mergesort($arr, $mid+1, $right);
            merge($arr, $left, $mid, $right);
        }
    }
    
    // This function merges the arrays
    function merge(&$arr, $left, $mid, $right) {

        $n1 = $mid - $left + 1; //LeftArray size
        $n2 = $right - $mid;     //RightArray size
        
        $LeftArray = array_fill(0, $n1, 0); 
        $RightArray = array_fill(0, $n2, 0);
        
        for($i=0; $i < $n1; $i++) { 
            $LeftArray[$i] = $arr[$left + $i];
        }
        for($i=0; $i < $n2; $i++) { 
            $RightArray[$i] = $arr[$mid + $i + 1];
        }
    

        $l=0; $r=0; $i=$left;
        while($l < $n1 && $r < $n2) {
            if($LeftArray[$l] < $RightArray[$r]) { 
                $arr[$i] = $LeftArray[$l]; 
                $l++; 
            }
            else { 
                $arr[$i] = $RightArray[$r];  
                $r++; 
            }
            $i++;
        }
    
        while($l < $n1) { 
            $arr[$i] = $LeftArray[$l];  
            $l++;  
            $i++;
        }
    
        while($r < $n2) { 
            $arr[$i] = $RightArray[$r]; 
            $r++;  
            $i++; 
        }
    }
    


    // --------------- Calling the function with array-------------
     
    $MyArray = array(100, 23, -10, 8, 0, 188, 928, 19, 989, 9, 1002);
    $n = sizeof($MyArray); 
    echo "Array:\n";
    print_r($MyArray);
    
    mergesort($MyArray, 0, $n-1);
    echo "Array After sorting:\n";
    print_r($MyArray);
?>